ui/desktop/src/utils/autoUpdater.ts

```diff
@@ -0,0 +1,493 @@
+import { autoUpdater, UpdateInfo } from 'electron-updater';
+import {
+  BrowserWindow,
+  ipcMain,
+  nativeImage,
+  Tray,
+  shell,
+  app,
+  dialog,
+  Menu,
+  MenuItemConstructorOptions,
+} from 'electron';
+import * as path from 'path';
+import * as fs from 'fs/promises';
+import log from './logger';
+import { githubUpdater } from './githubUpdater';
+import { loadRecentDirs } from './recentDirs';
+
+let updateAvailable = false;
+let trayRef: Tray | null = null;
+let isUsingGitHubFallback = false;
+let githubUpdateInfo: {
+  latestVersion?: string;
+  downloadUrl?: string;
+  releaseUrl?: string;
+  downloadPath?: string;
+  extractedPath?: string;
+} = {};
+
+// Store update state
+let lastUpdateState: { updateAvailable: boolean; latestVersion?: string } | null = null;
+
+// Configure auto-updater
+export function setupAutoUpdater(tray?: Tray) {
+  if (tray) {
+    trayRef = tray;
+  }
+
+  // Set the feed URL for GitHub releases
+  autoUpdater.setFeedURL({
+    provider: 'github',
+    owner: 'block',
+    repo: 'goose',
+    releaseType: 'release',
+  });
+
+  // Configure auto-updater settings
+  autoUpdater.autoDownload = false; // We'll trigger downloads manually
+  autoUpdater.autoInstallOnAppQuit = true;
+
+  // Set logger
```

Comment:
this helped me test this out locally - i couldn't get the update check to trigger without it

```suggestion
  autoUpdater.autoInstallOnAppQuit = true;

   // Enable updates in development mode for testing
   if (process.env.ENABLE_DEV_UPDATES === 'true') {
     autoUpdater.forceDevUpdateConfig = true;
   }
   
  // Set logger
```